arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/predef-test.kt

```diff
@@ -81,6 +80,46 @@ suspend fun assertCancellable(f: suspend () -> Unit): Unit {
   p.get() shouldBe ExitCase.Cancelled
 }
 
+suspend fun <A> Stream<A>.assertCancellable(): Unit {
+  val p = Promise<ExitCase>()
+  val start = Promise<Unit>()
+
+  val fiber = ForkAndForget {
+    guaranteeCase(
+      fa = {
+        Stream.effect { start.complete(Unit) }
+          .flatMap { this@assertCancellable }
+          .compile()
+          .drain()
+      },
+      release = { ex -> p.complete(ex) }
+    )
+  }
+
+  start.get()
+  fiber.cancel()
+  p.get() shouldBe ExitCase.Cancelled
+}
+
+@JvmName("assertStreamCancellable")
+suspend fun <A> assertCancellable(fa: (latch: Promise<Unit>) -> Stream<A>): Unit {
```

Comment:
If we would have this functions here as utility functions I think they should be generic enough to use them in more places and exclude the assertions from here, otherwise we're hiding the actual testing from the test classes.

Alternatively you could move them to the CancellationTest

Reply:
How would you make it more generic? You can still add your own assertions arround, as done [here](https://github.com/arrow-kt/arrow-fx/pull/232/files#diff-19c8e5cf480f27c6f5ec7dc44549de24R36). 

We can definitely move them to `CancellationTest`. We haven't discussed how we want to organise the test suite, now it's still mixed between a bunch of stuff in `StreamTest`, `ParJoinTest`, `BracketTest`, etc

I personally like grouping tests per combinator (or the `Concurrent` onces) so it can serve as a specification of it's behavior, like a law test suite.
If we'd do that I think we should add the `cancellation`/`interruption` tests there as well for it to be a full spec of it's behavior.

Reply:
@nomisRev ?